use/mobile-gaming-example.md

```diff
@@ -0,0 +1,540 @@
+
+Data arriving above the solid watermark line is _late data_—this is a score event that was delayed (perhaps generated offline) and arrived after the window to which it belongs had closed. Our pipeline's late-firing trigger ensures that this late data is still included in the sum.
+
+The following code example shows how `LeaderBoard` applies fixed-time windowing with the appropriate triggers to have our pipeline perform the calculations we want:
+
+```java
+// Extract team/score pairs from the event stream, using hour-long windows by default.
+@VisibleForTesting
+static class CalculateTeamScores
+    extends PTransform<PCollection<GameActionInfo>, PCollection<KV<String, Integer>>> {
+  private final Duration teamWindowDuration;
+  private final Duration allowedLateness;
+
+  CalculateTeamScores(Duration teamWindowDuration, Duration allowedLateness) {
+    this.teamWindowDuration = teamWindowDuration;
+    this.allowedLateness = allowedLateness;
+  }
+
+  @Override
+  public PCollection<KV<String, Integer>> apply(PCollection<GameActionInfo> infos) {
+    return infos.apply("LeaderboardTeamFixedWindows",
+        Window.<GameActionInfo>into(FixedWindows.of(teamWindowDuration))
+            // We will get early (speculative) results as well as cumulative
+            // processing of late data.
+            .triggering(AfterWatermark.pastEndOfWindow()
+                .withEarlyFirings(AfterProcessingTime.pastFirstElementInPane()
+                    .plusDelayOf(FIVE_MINUTES))
+                .withLateFirings(AfterProcessingTime.pastFirstElementInPane()
+                    .plusDelayOf(TEN_MINUTES)))
+            .withAllowedLateness(allowedLateness)
+            .accumulatingFiredPanes())
+        // Extract and sum teamname/score pairs from the event data.
+        .apply("ExtractTeamScore", new ExtractAndSumScore("team"));
+  }
+}
+```
+
+Taken together, these processing strategies let us address the latency and completeness issues present in the `UserScore` and `HourlyTeamScore` pipelines, while still using the same basic transforms to process the data—as a matter of fact, both calculations still use the same `ExtractAndSumScore` transform that we used in both the `UserScore` and `HourlyTeamScore` pipelines.
+
+## GameStats: Abuse Detection and Usage Analysis
+
+While `LeaderBoard` demonstrates how to use basic windowing and triggers to perform low-latency and flexible data analysis, we can use more advanced windowing techniques to perform more comprehensive analysis. This might include some calculations designed to detect system abuse (like spam) or to gain insight into user behavior. The `GameStats` pipeline builds on the low-latency functionality in `LeaderBoard` to demonstrate how you can use Beam to perform this kind of advanced analysis.
+
+Like `LeaderBoard`, `GameStats` reads data from a streaming source, in this case Pub/Sub. It is best thought of as an ongoing job that provides insight into the game as users play.
```

Comment:
streaming source -> unbounded source

(Streaming and Batch describe runtime modes. Bounded/unbounded describe data.)